server/sonar-server/src/test/java/org/sonar/server/organization/ws/SearchMyOrganizationsActionTest.java

```diff
@@ -0,0 +1,135 @@
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.organization.ws;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbTester;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.user.GroupDto;
+import org.sonar.db.user.UserDto;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.TestResponse;
+import org.sonar.server.ws.WsActionTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class SearchMyOrganizationsActionTest {
+  private static final String NO_ORGANIZATIONS_RESPONSE = "{\"organizations\": []}";
+
+  @Rule
+  public DbTester dbTester = DbTester.create(System2.INSTANCE);
+  @Rule
+  public UserSessionRule userSessionRule = UserSessionRule.standalone();
+
+  private DbClient dbClient = dbTester.getDbClient();
+
+  private WsActionTester underTest = new WsActionTester(new SearchMyOrganizationsAction(userSessionRule, dbClient));
+
+  @Test
+  public void returns_empty_response_when_user_is_not_logged_in() {
+    TestResponse response = underTest.newRequest().execute();
+
+    assertThat(response.getInput()).isEmpty();
+  }
+
+  @Test
+  public void returns_empty_array_when_user_is_logged_in_and_has_no_permission_on_anything() {
+    userSessionRule.logIn();
+
+    TestResponse response = underTest.newRequest().execute();
+
+    assertJson(response.getInput()).isSimilarTo(NO_ORGANIZATIONS_RESPONSE);
+  }
+
+  @Test
+  public void returns_organizations_of_authenticated_user_when_user_has_ADMIN_user_permission_on_some_organization() {
+    UserDto user = dbTester.users().insertUser();
+    dbTester.users().insertPermissionOnUser(dbTester.getDefaultOrganization(), user, SYSTEM_ADMIN);
+    OrganizationDto organization1 = dbTester.organizations().insert();
+    dbTester.users().insertPermissionOnUser(organization1, user, SYSTEM_ADMIN);
+    UserDto otherUser = dbTester.users().insertUser();
+    OrganizationDto organization2 = dbTester.organizations().insert();
+    dbTester.users().insertPermissionOnUser(organization2, otherUser, SYSTEM_ADMIN);
```

Comment:
you could also ensure that the permission "admin" is used by adding a different permission to `user` on ? organization2`.